src/main/resources/META-INF/plugin.xml

```diff
@@ -223,6 +223,12 @@
                 id="LSPDocumentLinkGotoDeclarationHandler"
                 implementation="com.redhat.devtools.lsp4ij.features.documentLink.LSPDocumentLinkGotoDeclarationHandler"/>
 
+        <!-- LSP textDocument/documentSymbol request support -->
+        <lang.psiStructureViewFactory
+                id="LSPDocumentSymbolStructureViewFactory"
+                language="textmate"
```

Comment:
It will work only with textmate. We need to create those factory with Java code.